File: week_10/c_sharp_course/week_1/day_4/composition/checkpoint_1_composition/wizard_management/Wizard.cs
using Behaviours;

namespace WizardManagement
{
  public class Wizard
  {
    private string name;
    private IFly ride;

    public string Name 
    {
      get { return name; }
      set { name = value; }
    }

    public IFly Ride
    {
      get { return ride; }
      set { ride = value; }
    }
    
    public Wizard(string name, IFly ride)
    {
      this.name = name;
      this.ride = ride;
    }

    public string Fly()
    {
      return this.ride.Fly();
    }
  }
}

File: week_06/day_3/composition/Composition_Lesson_End/app/src/main/java/example/codeclan/com/composition_lesson_end/wizard_management/Wizard.java
package example.codeclan.com.composition_lesson_end.wizard_management;

import example.codeclan.com.composition_lesson_end.behaviours.Flyable;

/**
 * Created by user on 28/08/2017.
 */

public class Wizard {
    String name;
    Flyable ride;

    public Wizard(String name, Flyable ride){
        this.name = name;
        this.ride = ride;
    }

    public void setRide(Flyable ride){
        this.ride = ride;
    }

    public String getName(){
        return this.name;
    }

    public Flyable getRide(){
        return this.ride;
    }

    public String fly(){
        return this.ride.fly();
    }

}
